### ISSUE TITLE
macOS 11.5 causes finder to fail

### ISSUE THREAD

**Issue by nick-at-artsed** [targendaz2/Mac-Set-Default-Apps#4]
Using the examples on macOS 11.5 leaves Finder unable to read and write. Example: The Finder can't complete the operation because some data in "somefile.xlsx" can't be read or written. (Error code -36)
Using `msda set com.apple.safari -p http -p https -u public.url all -u public.html viewer -u public.xhtml all` will return functionality.

**Comment by targendaz2** [comment]
@nick-at-artsed Could you provide the exact steps to reproduce this?

**Comment by nick-at-artsed** [comment]
@targendaz2 yes this is on either a mac mini M1 or MacBook Air M1 on macOS 11.5

The issue occurs when...

- Running msda.py as a Jamf Pro script / policy with JAMF variable as True and...
- Parameter 4: `set com.microsoft.edgemac -p http -p https -u public.url all -u public.html viewer -u public.xhtml all`
or
- Parameter 4: `set com.microsoft.outlook -p mailto -p webcal -u com.apple.ical.ics all -u com.apple.ical.vcs all`

I can also replicate the issue by...

- Installing MacSetDefaultApps-v1.3.1.pkg
- `msda set com.microsoft.edgemac -p http -p https -u public.url all -u public.html viewer -u public.xhtml all`
or
- `msda set com.microsoft.outlook -p mailto -p webcal -u com.apple.ical.ics all -u com.apple.ical.vcs all`

In either scenario I can restore Finders normal functionality with `msda set com.apple.safari -p http -p https -u public.url all -u public.html viewer -u public.xhtml all` or by changing parameter 4 to `com.apple.safari`

**Comment by targendaz2** [comment]
@nick-at-artsed Thanks for that info. If you could give me the exact steps to test this as well, I can do some troubleshooting.

**Comment by nick-at-artsed** [comment]
@targendaz2 I have made a screen recording for you here: 
[https://artsed001.blob.core.windows.net/assets/msda_finder.mov](url)

**Comment by targendaz2** [comment]
@nick-at-artsed It looks like you're not running the commands as root (i.e. with sudo). Does the same behavior happen if you do that?

**Comment by nick-at-artsed** [comment]
@targendaz2 yes, the same behaviour when run with sudo privilege, also when ran as a Jamf Pro script (ie., also root). 
Our devices use Jamf Connect so am wondering if this changes anything about lshandlers? I will try to replicate on a clean install to eliminate this.

**Comment by eduder** [comment]
We have seen this behavior intermittently on our Macs with Big Sur since deploying MSDA, however until seeing this post I never thought the two might be related. Would be willing to help troubleshoot to get further details on one of our affected machines if it will help.

**Comment by targendaz2** [comment]
I'm working on v2 of this app under the `jxa-v2` branch. It should resolve this issue. I don't have an ETA at this time, but I'll follow up once it's released.
